### ISSUE TITLE
[Syncfusion Flutter Charts]  Panning and zooming causes the isVisible property of all series to true

### ISSUE THREAD

**Issue by chaudharydeepanshu** [syncfusion/flutter-widgets#839]
When panning and zooming the chart the isVisible property of the series becomes true even if it was set false before.

Sample video showing the issue:-

https://user-images.githubusercontent.com/85361211/181842580-fdfe09ef-d7c3-4f65-9ea9-553c38c35f75.mp4

Sample code for the chart widget :-

```Dart
class ChartWidget extends StatelessWidget {
  const ChartWidget(
      {Key? key,})
      : super(key: key);

  @override
  Widget build(BuildContext context) {
    final bool y1ChartSeries = false;
    final bool y2ChartSeries = false;
    final bool y3ChartSeries = false;
    final bool y4ChartSeries = false;
    final List<Color> chartSeriesPalette = <Color>[
      Colors.red,
      Colors.teal,
      Colors.orange,
      Colors.brown,
    ];
    final List<ChartData> chartData = [
      ChartData(x: DateTime(2022, 7, 24), y1: 4, y2: 3, y3: 4, y4: 10, y: 21),
      ChartData(x: DateTime(2022, 7, 25), y1: 13, y2: 6, y3: 10, y4: 1, y: 30),
      ChartData(x: DateTime(2022, 7, 26), y1: 0, y2: 0, y3: 0, y4: 0, y: 0),
      ChartData(x: DateTime(2022, 7, 27), y1: 2, y2: 2, y3: 15, y4: 35, y: 54),
      ChartData(x: DateTime(2022, 7, 28), y1: 15, y2: 9, y3: 5, y4: 13, y: 42),
      ChartData(x: DateTime(2022, 7, 29), y1: 10, y2: 12, y3: 6, y4: 7, y: 35),
      ChartData(x: DateTime(2022, 7, 30), y1: 20, y2: 5, y3: 7, y4: 16, y: 49),
    ];
    return Center(
      child: SfCartesianChart(
        tooltipBehavior: TooltipBehavior(enable: true),
        zoomPanBehavior: ZoomPanBehavior(
          enablePinching: true,
          zoomMode: ZoomMode.x,
          enablePanning: true,
        ),
        primaryXAxis: DateTimeCategoryAxis(
          interval: chartData.length < 8 ? 1 : null,
          dateFormat: chartData.length < 8
              ? DateFormat.E()
              : chartData.length < 32
                  ? DateFormat.MMMMd()
                  : DateFormat.MMMMd(),
        ),
        series: <ChartSeries>[
          AreaSeries<ChartData, DateTime>(
            name: "All Tasks",
            gradient: LinearGradient(
              begin: Alignment.topCenter,
              end: Alignment.bottomCenter,
              colors: [
                Theme.of(context).colorScheme.onSurface,
                Theme.of(context).colorScheme.onSurface.withOpacity(0.05),
              ],
              stops: const [0.0, 1.0],
            ),
            dataSource: chartData,
            xValueMapper: (ChartData data, _) => data.x,
            yValueMapper: (ChartData data, _) => data.y,
            dataLabelSettings:
                DataLabelSettings(isVisible: true, showZeroValue: false),
            markerSettings:
                MarkerSettings(isVisible: true, height: 3, width: 3),
          ),
          LineSeries<ChartData, DateTime>(
            name: "Reading Tasks",
            color: chartSeriesPalette[0],
            isVisible: y1ChartSeries,
            dataSource: chartData,
            xValueMapper: (ChartData data, _) => data.x,
            yValueMapper: (ChartData data, _) => data.y1,
            markerSettings:
                MarkerSettings(isVisible: true, height: 3, width: 3),
          ),
          LineSeries<ChartData, DateTime>(
            name: "Flashcards Tasks",
            color: chartSeriesPalette[1],
            isVisible: y2ChartSeries,
            dataSource: chartData,
            xValueMapper: (ChartData data, _) => data.x,
            yValueMapper: (ChartData data, _) => data.y2,
            markerSettings:
                MarkerSettings(isVisible: true, height: 3, width: 3),
          ),
          LineSeries<ChartData, DateTime>(
            name: "Matches Tasks",
            color: chartSeriesPalette[2],
            isVisible: y3ChartSeries,
            dataSource: chartData,
            xValueMapper: (ChartData data, _) => data.x,
            yValueMapper: (ChartData data, _) => data.y3,
            markerSettings:
                MarkerSettings(isVisible: true, height: 3, width: 3),
          ),
          LineSeries<ChartData, DateTime>(
            name: "MCQ Quiz Tasks",
            color: chartSeriesPalette[3],
            isVisible: y4ChartSeries,
            dataSource: chartData,
            xValueMapper: (ChartData data, _) => data.x,
            yValueMapper: (ChartData data, _) => data.y4,
            markerSettings:
                MarkerSettings(isVisible: true, height: 3, width: 3),
          ),
        ],
      ),
    );
  }
}
```

So, is this the desired behavior of zooming or panning?

Thank you.

**Comment by Yuvaraj-Gajaraj** [comment]
Hi @chaudharydeepanshu,

We were able to reproduce the reported issue and logged a feedback request. You can track the status of this issue using the following link.

Feedback: [https://www.syncfusion.com/feedback/36732](https://www.syncfusion.com/feedback/36732/panning-and-zooming-cause-the-series-visibility)

Regards,
Yuvaraj.

**Comment by chaudharydeepanshu** [comment]
Thanks for letting me know. Looking forward to the fix.

**Comment by LavanyaGowtham2021** [comment]
Hi @chaudharydeepanshu, 
 
We would like to let you know that the reported issue "Hidden series becomes visible while zooming and panning the chart" has been fixed and rolled out in our 11th October 2022 weekly Nuget. So, we kindly request that you to upgrade the SfCartesianChart package to the latest version below. 
 
https://pub.dev/packages/syncfusion_flutter_charts/versions/20.3.49  
 
Regards, 
Lavanya A.

**Comment by chaudharydeepanshu** [comment]
Thanks for letting me know 👍. I can verify that this issue has been fixed in version [20.3.49](https://pub.dev/packages/syncfusion_flutter_charts/versions/20.3.49).
